src/core/default_appearance.js

```diff
@@ -87,6 +87,94 @@ function parseDefaultAppearance(str) {
   return new DefaultAppearanceEvaluator(str).parse();
 }
 
+class AppearanceEvaluator extends EvaluatorPreprocessor {
+  constructor(str) {
+    super(str);
+    this.str = str;
+  }
```

Comment:
Let's be explicit about this method expecting a *stream*, since we most of the time use "str" as a shorthand for "string" in the code-base:
```suggestion
  constructor(stream) {
    super(stream);
    this.stream = stream;
  }
```